### PR TITLE
getPolicy should return `ErrNoPolicy` when `o.Store.Get` fails

### PR DESCRIPTION
**Pull request by dreemkiller**
This is a fix for #25

### CODE REVIEW

**Comment by dreemkiller** [comment]
It's possible that we also want to remove:
```
	if len(vals) == 0 {
		return nil, ErrNoPolicy
```
below this change, as `o.Store.Get` will never return a `vals` with length `0`

File: vts/policymanager/policymanager.go [hunk 1]
```diff
@@ -67,7 +67,7 @@ func (o *PolicyManager) getPolicy(ev *proto.EvidenceContext) (*policy.Policy, er
 
 	vals, err := o.Store.Get(policyID)
 	if err != nil {
-		return nil, err
+		return nil, ErrNoPolicy
```
Comment (thomas-fossati) [inline vts/policymanager/policymanager.go:70]: the only problem is we lose information about the root cause.

what about:
```suggestion
		return nil, fmt.Errorf("%w: %v", ErrNoPolicy, err)
```

And if the caller needs to branch based on the error type they still can do that:
```
        if errors.Is(err, ErrNoPolicy) {
                // specific handling for "no policy"
        } else {
                // generic error handling
        }
```

Comment (dreemkiller) [reply vts/policymanager/policymanager.go:70]: That should work

**Review by setrofim — CHANGES_REQUESTED** [review]
Unfortunately, this solution won't work as `ErrNoPolicy`  is meant expressly to indicate that no error occurred in retrieval, but the policy doesn't exist. It gets "swallowed" here: https://github.com/veraison/services/blob/main/vts/policymanager/policymanager.go#L44

If any error other than "not found" occurs, we want to propagate it, not swallow it. So we don't want to wrap all errors returned from `Get()` as `ErrNotFound`.

I made the wrong assumption about KVStore API. I think the correct fix might be to push this down into KVStore, and have `IKVStore.Get()` return `ErrKeyDoesNotExist` or something.

I'll look into it.

**Comment by dreemkiller** [comment]
I was afraid that modifying `IKVStore.Get()` might be the best answer.

**Comment by yogeshbdeshpande** [comment]
> Unfortunately, this solution won't work as `ErrNoPolicy` is meant expressly to indicate that no error occurred in retrieval, but the policy doesn't exist. It gets "swallowed" here: https://github.com/veraison/services/blob/main/vts/policymanager/policymanager.go#L44
> 
> If any error other than "not found" occurs, we want to propagate it, not swallow it. So we don't want to wrap all errors returned from `Get()` as `ErrNotFound`.
> 
> I made the wrong assumption about KVStore API. I think the correct fix might be to push this down into KVStore, and have `IKVStore.Get()` return `ErrKeyDoesNotExist` or something.
> 
> I'll look into it.

We should raise a new github issue to fix it properly in KVStore and close this PR referring to it, then!

**Comment by setrofim** [comment]
This is superseded by https://github.com/veraison/services/pull/30

**Pull request closed without merging**